### PR TITLE
upgrade the sfdx-lwc-jest to latest

### PR DESCRIPTION
**Pull request by msrivastav13**
### What does this PR do?

upgrade the sfdx-lwc-jest to latest.

Note: This gets latest from master to peters API version upgrade branch.

Updated jest timeout to 2X since few tests timeout.

### What issues does this PR fix or reference?

#<Insert GitHub Issue>

## The PR fulfills these requirements:

[ ] Tests for the proposed changes have been added/updated.
[ ] Code linting and formatting was performed.

### Functionality Before

<insert gif and/or summary>

### Functionality After

<insert gif and/or summary>

### CODE REVIEW

**Comment by msrivastav13** [comment]
@pchittum / @albarivas I was trying to update the `sfdx-lwc-jests` for v50.0. Since @pchittum branch had all other v50.0 changes I create a PR to peters api version branch.

The branch was behind master by quite a lot of commits so you see a gigantic one.

Scratch org creation I guess is failing because may be devhub is on  v49.0 version still.

Anyways i want to leave here so you can see changes we need in jest.config file as per the https://github.com/salesforce/sfdx-lwc-jest/issues/175

File: jest.config.js [hunk 1]
```diff
@@ -1,4 +1,6 @@
 const { jestConfig } = require('@salesforce/sfdx-lwc-jest/config');
+const setupFilesAfterEnv = jestConfig.setupFilesAfterEnv || [];
```
Comment (albarivas) [inline jest.config.js:2]: I'm curious about, why did you need this change?

Comment (albarivas) [reply jest.config.js:2]: never mind, I've just read why 👍
